UT - resolver " FAIL: TestSolveOperators_WithSystemConstraints"

## Bug Report
Unit test - resolver_test.go  " FAIL: TestSolveOperators_WithSystemConstraints"
https://github.com/operator-framework/operator-lifecycle-manager/runs/4756877867?check_suite_focus=true#step:5:53

```
--- FAIL: TestSolveOperators_WithSystemConstraints (0.01s)
    resolver_test.go:164: 
        	Error Trace:	resolver_test.go:164
        	Error:      	"constraints not satisfiable: package: packageC is not white listed, package: packageB is not white listed, subscription packageA-alpha requires test-catalog/test-namespace/alpha/packageA.v1, subscription packageA-alpha exists, bundle packageA.v1 requires an operator providing an API with group: g, version: v, kind: k" does not contain "packageA is not white listed"
        	Test:       	TestSolveOperators_WithSystemConstraints
        	Messages:   	Test Runtime constraints don't allow A failed
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [packageA.v1]"
time="2022-01-10T03:41:22Z" level=warning msg="could not unambiguously infer package name for \"packageA.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [packageA.v1]"
time="2022-01-10T03:41:22Z" level=warning msg="could not unambiguously infer package name for \"packageA.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [packageA.v1]"
time="2022-01-10T03:41:22Z" level=warning msg="could not unambiguously infer package name for \"packageA.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [packageA.v1]"
time="2022-01-10T03:41:22Z" level=warning msg="could not unambiguously infer package name for \"packageA.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [packageA.v1]"
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [packageA.v1]"
time="2022-01-10T03:41:22Z" level=warning msg="could not unambiguously infer package name for \"packageA.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [packageA.v1]"
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [packageA.v1]"
time="2022-01-10T03:41:22Z" level=warning msg="could not unambiguously infer package name for \"packageA.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [packageA.v1]"
time="2022-01-10T03:41:22Z" level=warning msg="could not unambiguously infer package name for \"packageA.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [packageA.v1]"
time="2022-01-10T03:41:22Z" level=warning msg="could not unambiguously infer package name for \"packageA.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [packageA.v1]"
time="2022-01-10T03:41:22Z" level=warning msg="could not unambiguously infer package name for \"packageA.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [packageA.v1]"
time="2022-01-10T03:41:22Z" level=info msg="considered csvs without properties annotation during resolution: [opA.v1.0.0 opB.v1.0.0]"
time="2022-01-10T03:41:23Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1]"
time="2022-01-10T03:41:24Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1]"
constraints not satisfiable: subscription a-alpha requires catsrc/catsrc-namespace/alpha/a.v1, subscription a-alpha exists, catsrc/catsrc-namespace/alpha/a.v1 and catsrc/catsrc-namespace/beta/a.v2 originate from package a, subscription a-beta requires catsrc/catsrc-namespace/beta/a.v2, subscription a-beta exists
time="2022-01-10T03:41:24Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1]"
time="2022-01-10T03:41:24Z" level=warning msg="could not unambiguously infer package name for \"a.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:24Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1]"
time="2022-01-10T03:41:24Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1]"
time="2022-01-10T03:41:24Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1]"
time="2022-01-10T03:41:24Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1]"
time="2022-01-10T03:41:25Z" level=warning msg="could not unambiguously infer package name for \"a.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:25Z" level=warning msg="could not unambiguously infer package name for \"b.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:25Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1 b.v1]"
time="2022-01-10T03:41:25Z" level=warning msg="could not unambiguously infer package name for \"a.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:25Z" level=warning msg="could not unambiguously infer package name for \"b.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:25Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1 b.v1]"
time="2022-01-10T03:41:25Z" level=warning msg="could not unambiguously infer package name for \"a.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:25Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1]"
time="2022-01-10T03:41:25Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1 b.v1]"
time="2022-01-10T03:41:25Z" level=warning msg="could not unambiguously infer package name for \"a.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:25Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1]"
time="2022-01-10T03:41:25Z" level=warning msg="could not unambiguously infer package name for \"a.v1\" (found 0 distinct package names)"
time="2022-01-10T03:41:25Z" level=info msg="considered csvs without properties annotation during resolution: [a.v1]"
FAIL
FAIL	github.com/operator-framework/operator-lifecycle-manager/pkg/controller/registry/resolver	3.763s

```
<!-- 
Note: Make sure to first check the prerequisites that can be found in the main README file!

Thanks for filing an issue! Before hitting the button, please answer these questions.
Fill in as much of the template below as you can. If you leave out information, we can't help you as well.
-->

**What did you do?**
CI Unit test intermittently fails at  TestSolveOperators_WithSystemConstraints.

**What did you expect to see?**
No failure

**What did you see instead? Under which circumstances?**
Failure above.


**Environment**
* operator-lifecycle-manager version:

<!--- Insert operator-lifecycle-manager release or Git SHA here. -->

* Kubernetes version information:

<!--- Insert the output of `kubectl version` here -->

* Kubernetes cluster kind: 

**Possible Solution**
<!--- Only if you have suggestions on a fix for the bug -->

**Additional context**
Add any other context about the problem here.

--- a/pkg/controller/registry/resolver/resolver_test.go
+++ b/pkg/controller/registry/resolver/resolver_test.go
@@ -140,7 +140,7 @@ func TestSolveOperators_WithSystemConstraints(t *testing.T) {
 		{
 			title:                     "Runtime constraints don't allow A",
 			snapshotEntries:           []*cache.Entry{packageA, packageB, packageC, packageD},
-			systemConstraintsProvider: whiteListConstraintProvider(),
+			systemConstraintsProvider: whiteListConstraintProvider(packageB, packageC, packageD),
 			expectedOperators:         nil,
 			csvs:                      nil,
 			subs:                      []*v1alpha1.Subscription{packageASub},

